docs/user_guide/event_processors/event_starlark.md

```diff
@@ -0,0 +1,386 @@
+### Intro
+
+The `event-starlark` processor applies a [`Starlark`](https://github.com/google/starlark-go/blob/master/doc/spec.md) function on a list of `event` messages before returning them to the processors pipeline and then to the output.
+
+`starlark` is a dialect of Python, developed initially for the [Bazel build tool](https://bazel.build/) but found multiple uses as a configuration language embedded in a larger application.
+
+There are a few differences between Python and Starlark, programs written in Starlark are supposed to be short-lived and have no external side effects, their main result is structured data or side effects on the host application. As a result, Starlark has no need for classes, exceptions, reflection, concurrency, and other such features of Python.
+
+`gNMIc` uses the Go implementation of Starlark, the language definition can be found [here](https://github.com/google/starlark-go/blob/master/doc/spec.md)
+
+A Starlark program running as a `gNMIc` processor should define an `apply` function that takes an arbitrary number of arguments of type `Event` and returns zero or more `Event`s.
+
```

Comment:
I think the question I have here is "On what event (not a gnmic even) a starlark processor would be triggered?"

In which case the list of events passed to the `apply` function will be >1?

Reply:
The starlark processor is not different from other processors, if it is defined under an output it will be applied to the events the output is writing out.
The number of events passed to a processor (starlark or not) depends on the number of notifications in a single subscribe response and if events were merged by another processor earlier in the pipeline.